src/main/java/io/stargate/sgv2/jsonapi/service/shredding/Shredder.java

```diff
@@ -88,14 +88,21 @@ public WritableShreddedDocument shred(JsonNode doc, UUID txId, DocumentProjector
     // And then we can validate the document size
     validateDocumentSize(documentLimits, docJson);
 
-    // Before shredding, may need to drop "non-indexed" fields:
+    final WritableShreddedDocument.Builder b =
```

Comment:
We need document builder immediately, to build index incrementally: in particular, `$vector` indexing should precede "no index" Projection that would typically drop it.

Reply:
Any reason why it needs to be if `allow` is specified, can't we set up required indexing fields with `$vector` in it?

Reply:
Will proceed with this idea, change code here.